api/v1/l7_logcollector_types.go

```diff
@@ -0,0 +1,173 @@
+// Copyright (c) 2024 Tigera, Inc. All rights reserved.
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+)
+
+// L7LogCollectorDaemonSet is the configuration for the L7LogCollector DaemonSet.
+type L7LogCollectorDaemonSet struct {
+
+	// Spec is the specification of the L7LogCollector DaemonSet.
+	// +optional
+	Spec *L7LogCollectorDaemonSetSpec `json:"spec,omitempty"`
+}
+
+// L7LogCollectorDaemonSetSpec defines configuration for the L7LogCollector DaemonSet.
+type L7LogCollectorDaemonSetSpec struct {
+
+	// Template describes the L7LogCollector DaemonSet pod that will be created.
+	// +optional
+	Template *L7LogCollectorDaemonSetPodTemplateSpec `json:"template,omitempty"`
+}
+
+// L7LogCollectorDaemonSetPodTemplateSpec is the L7LogCollector DaemonSet's PodTemplateSpec
+type L7LogCollectorDaemonSetPodTemplateSpec struct {
+
+	// Spec is the L7LogCollector DaemonSet's PodSpec.
+	// +optional
+	Spec *L7LogCollectorDaemonSetPodSpec `json:"spec,omitempty"`
+}
+
+// L7LogCollectorDaemonSetPodSpec is the L7LogCollector DaemonSet's PodSpec.
+type L7LogCollectorDaemonSetPodSpec struct {
+	// InitContainers is a list of L7LogCollector DaemonSet init containers.
+	// If specified, this overrides the specified L7LogCollector DaemonSet init containers.
+	// If omitted, the L7LogCollector DaemonSet will use its default values for its init containers.
+	// +optional
+	InitContainers []L7LogCollectorDaemonSetInitContainer `json:"initContainers,omitempty"`
+
+	// Containers is a list of L7LogCollector DaemonSet containers.
+	// If specified, this overrides the specified L7LogCollector DaemonSet containers.
+	// If omitted, the L7LogCollector DaemonSet will use its default values for its containers.
+	// +optional
+	Containers []L7LogCollectorDaemonSetContainer `json:"containers,omitempty"`
+}
+
+// L7LogCollectorDaemonSetContainer is a L7LogCollector DaemonSet container.
+type L7LogCollectorDaemonSetContainer struct {
+	// Name is an enum which identifies the L7LogCollector DaemonSet container by name.
+	// +kubebuilder:validation:Enum=l7-collector;envoy-proxy;dikastes
```

Comment:
This would be the right time to make our resource names all consistent.
How about we rename the daemonset from `l7-log-collector` to `l7-collector`? (Or vice versa).  Then all of these names (container, deploy, crd) are consistent.
@caseydavenport @tmjd

Reply:
Does the DaemonSet name matter at all? The DaemonSet name doesn't appear in our API at all, and changing it would require us to explicitly delete the old one IIUC. This would be easy to do, but not sure if there's a compelling reason to do it?

Reply:
Many structs in this file are called `L7LogCollectorDaemonSetX` which is taken from the daemonset name, so it is but a matter of how you want to look at it.

The daemon set isn't only doing collection, it has envox-proxy and dikastes. Perhaps L7DaemonSetX would be just as fitting for the structs in this file?

Reply:
After discussing offline, we decided not to modify the deployment.